shared/src/main/scala/coop/rchain/dag/DagOps.scala

```diff
@@ -0,0 +1,158 @@
+      tipAncestors <- (Seq(tip), Seq(tip)).tailRecM[F, Seq[A]] {
+                       case (acc, blocksToProceed) =>
+                         blocksToProceed.toList match {
+                           case Nil =>
+                             acc.asRight[(Seq[A], Seq[A])].pure[F]
+                           case blocks =>
+                             for {
+                               nextBlocks <- blocks
+                                              .traverse(
+                                                b =>
+                                                  dag
+                                                    .parents(b)
+                                                    .map(_.getOrElse(Set.empty))
+                                              )
+                                              .map(_.flatten)
+                               // stop when base block is met
+                               next = if (nextBlocks.contains(base)) Seq.empty else nextBlocks
+                             } yield (acc ++ next, next).asLeft[Seq[A]]
+                         }
+                     }
+      // blocks from tipSet connected to base
+      branchSet <- (Seq.empty[A], Seq(base))
+                    .tailRecM[F, Seq[A]] {
+                      case (acc, blocksToProceed) =>
+                        blocksToProceed.toList match {
+                          case Nil =>
+                            acc.asRight[(Seq[A], Seq[A])].pure[F]
+                          case blocks =>
+                            for {
+                              children <- blocks
+                                           .traverse(
+                                             b =>
+                                               dag
+                                                 .children(b)
+                                                 .map(_.getOrElse(Seq.empty))
+                                           )
+                                           .map(_.flatten)
+                              childrenFromTipSet = children.filter(tipAncestors.contains)
+                            } yield (acc ++ childrenFromTipSet, childrenFromTipSet).asLeft[Seq[A]]
+                        }
+                    }
+    } yield (tip, branchSet)
+
+  /**
+    * @return seq of branches, sorted by size starting with the biggest one
+    */
+  def computeSortedBranches[F[_]: Concurrent, A](
+      tips: Seq[A],
+      base: A,
+      dag: DagReader[F, A]
+  )(implicit ord: Ordering[A]): F[Seq[(A, Seq[A])]] =
+    for {
+      // compute branches in parallel, sort according to size primarily and tip secondary, in reverse order
+      branches <- fs2.Stream
+                   .emits(tips.map { top =>
+                     fs2.Stream.eval(computeBranch(top, base, dag))
+                   })
+                   .parJoinUnbounded
+                   .compile
+                   .toList
+                   .map(_.sortBy(b => (b._2.size, b._1)))
```

Comment:
The sorting rules should be a core part of the consensus protocol. I am OK to use the rule here right now. But we should probably discuss more on this(Maybe on https://github.com/rchain/rchip-proposals/issues/23 or should we start a fresh new one totally on consensus part?).

Reply:
We can calculate total stake on `b._2` to sort.

Reply:
@zsluedem I agree, yes, how branches are constructed should be part of a protocol, as different order lead to different hashes. Getting stake involved is an interesting idea, but this will create incentive to put deploys into validator with the most stake to get smaller reject rate, which is not what we want, imo.